src/agent.cpp

```diff
@@ -1070,8 +1116,11 @@ bool Agent::doDefendGoal() {
 
 bool Agent::doGoToBall() {
   const WorldModel & wm = this->world();
-  Body_GoToPoint(wm.ball().pos(), 0.25, ServerParam::i().maxDashPower()).execute(this);
-  return true;
+  const BallObject& ball = wm.ball();
+  if (! ball.rposValid()) {
```

Comment:
Good point; correcting. Admittedly, if one used a doMove and/or doTurn, the relative ball position could be used. As it is, I find it questionable why Intercept is considered a mid-level action and Go_To_Ball is considered a high-level action, and not the reverse.